src/main/java/picard/sam/SamToFastqWithTags.java

```diff
@@ -0,0 +1,213 @@
+package picard.sam;
+
+import htsjdk.samtools.SAMReadGroupRecord;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.fastq.FastqRecord;
+import htsjdk.samtools.fastq.FastqWriter;
+import htsjdk.samtools.fastq.FastqWriterFactory;
+import htsjdk.samtools.util.IOUtil;
+import htsjdk.samtools.util.Log;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import picard.PicardException;
+import picard.cmdline.programgroups.ReadDataManipulationProgramGroup;
+
+import java.io.File;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@CommandLineProgramProperties(
+        summary = SamToFastqWithTags.USAGE_SUMMARY + SamToFastqWithTags.USAGE_DETAILS,
+        oneLineSummary = SamToFastqWithTags.USAGE_SUMMARY,
+        programGroup = ReadDataManipulationProgramGroup.class)
+public class SamToFastqWithTags extends SamToFastq {
+    static final String USAGE_SUMMARY = "Converts a SAM or BAM file to FASTQ.";
```

Comment:
Should its summary be the same as the one in `SamToFastq`?

Reply:
whoops